tests/unit/stall_detector_test_utilities.hh

```diff
@@ -91,6 +91,7 @@ void spin_some_cooperatively(std::chrono::duration<double> how_much, void_fn bod
     }
 }
 
+[[maybe_unused]]
```

Comment:
does maybe_unsued imply inline? inline should be there in any case right?

Reply:
No, I don't see why it would imply it?

I don't think it should be inline: do you mean to adjust the linkage? It's already in an anon namespace, so it has internal linkage.

An earlier version of this had `inline` "by mistake", both work to disable the warning. I think it's just some kind of heuristic on the part of clang (and gcc). See for example this question: https://stackoverflow.com/questions/47542751/gcc-warns-about-unused-static-functions-but-not-static-inline-is-there-a-pract.

Reply:
I chose maybe_unused over inline just because I didn't have any reason to want the function to be inlined.

Happy to change it too.

Reply:
Ah ok yeah that's fine then. I missed this bit:

> It's already in an anon namespace, so it has internal linkage.